lib/analyse.ml

```diff
@@ -191,7 +195,14 @@ module Analysis = struct
             Current.Process.check_output ~cwd:dir ~cancellable:true ~job cmd >>= begin function
               | Error _ ->
                 (* new release *)
-                Lwt_result.return (add_pkg ~path ~name ~package New pkgs)
+                let package_dir = "packages/"^name in
+                let cmd = "", [| "git"; "show"; master^":"^package_dir |] in
+                Current.Process.check_output ~cwd:dir ~cancellable:true ~job cmd >>= begin function
+                | Ok _ -> Lwt_result.return (add_pkg ~path ~name ~package NewRelease pkgs)
+                | Error _ ->
+                  (* newly published package *)
+                  Lwt_result.return (add_pkg ~path ~name ~package NewPackage pkgs)
```

Comment:
Since all we need to compute here is the kind of change, it think it could arguably be cleaner if this were something like (incorporating both my and Mark's previous suggestions):

```suggestion
                let+ is_newly_published = is_newly_published_package ~cwd ~job name master in
                let change = if is_newly_published then Package else Release in
                add_pkg ~path ~name ~package change_kind (New change)
```

This is arguably stylistic, so I wouldn't insist here. But I find these very long, deeply nested functions pretty hard to read compared to flatter alternatives, so I'm concerned with maintainability not just my fashion sense :)

Reply:
We are currently using infix syntax for Lwt map/bind and `let+`, `let*`, etc for OCurrent. So, though stylistically I like your suggestion, this may be confusing. I've incorporated the change partially, still keeping the nested function.

Reply:
That makes sense!